src/nupic/ntypes/Sdr.hpp

```diff
@@ -748,32 +795,201 @@ class SparseDistributedRepresentation : public Serializable
         string marker;
         UInt version;
         inStream >> marker >> version;
-        NTA_ASSERT( marker == "SDR" );
-        NTA_ASSERT( version == SERIALIZE_VERSION );
+        NTA_CHECK( marker == "SDR" );
```

Comment:
add check `NTA_CHECK(callbacks.size() == 0) << "SDR cannot serialize with callbacks assigned. Remove all callbacks before serialization."; `

Reply:
I don't want to force the user to disassemble their program before saving their data.  What if they're still using the program and just want to save a snapshot of it running?

I understand that save/load results in a different & incomplete result, but I think that this is a problem which the caller needs to sort out, since they're the ones who set it up and their setup is irreproducible from the SDR's side of things.  I will add some documentation describing these shortcomings.